### PR TITLE
Deprecated parameters in several annotations are ignored - inconsisten

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/4097

### CODE REVIEW

**Comment by ppalaga** [comment]
I am perhaps missing something, but why should we add support for deprecated stuff?
The supported cases are documented in https://camel.apache.org/camel-quarkus/2.12.x/user-guide/cdi.html#_endpointinject_and_produce

**Comment by llowinge** [comment]
@ppalaga Problem is as follows. Using

```
@EndpointInject(id = "mock:result")
    protected MockEndpoint resultEndpoint;

    @Produce(id = "direct:start")
    protected ProducerTemplate template;
 ``` fails with `java.lang.IllegalArgumentException: uri must be specified and not empty.`. I'm not sure if it is bug in Camel or CQ. But even using deprecated stuff shouldn't break it IMO. I think if some feature is deprecated + it doesn't work, it should be removed.

**Comment by JiriOndrusek** [comment]
From my PoV support of deprecated parameters is not worth of a fix - I agree. But the stat of the code is, that user may use `@Consume` with deprecated parameter, but if `@Produce` or `@EndpountInject` is used with deprecated parameter, execution fails. The doc from the only says, that the same approach as in plain Camel can used -but plain Caml allows deprecated parameters.

**Comment by ppalaga** [comment]
> fails with java.lang.IllegalArgumentException: uri must be specified and not empty.. I'm not sure if it is bug in Camel or CQ. But even using deprecated stuff shouldn't break it IMO. I think if some feature is deprecated + it doesn't work, it should be removed.

The evaluation of the annotation params happens (or is missing in) in Camel Quarkus. I'd vote for throwing a better error message (something like `@org.apache.camel.EndpointInject(uri = "my-component:...") is not supported on Camel Quarkus. Please replace it with just @org.apache.camel.EndpointInject("my-component:...")`. We could output a similar warning for `@Consume`. WDYT?

**Comment by JiriOndrusek** [comment]
I agree with that approach and will change the PR accordingly. 

@llowinge Do you agree?

**Comment by llowinge** [comment]
@JiriOndrusek I agree with @ppalaga.

**Comment by JiriOndrusek** [comment]
PR is changed to throw an IllegalArgumentException with appropriate message + tests covering these use cases are added as well.

File: extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedConsumeTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    static final QuarkusUnitTest CONFIG = new QuarkusUnitTest().setExpectedException(IllegalArgumentException.class)
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class));
+
+    @Consume(uri = "direct:consumeAnnotation")
+    public String consumeAnnotation(String name) {
+        return "Consumed " + name;
+    }
+
+    @Test
+    public void camelMainEventBridgeNotConfigured() {
```
Comment (jamesnetherton) [inline extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedConsumeTest.java:37]: If the empty test is required to make the tests run, please can we name it something relevant to the thing under test. E.g:

```
    @Test
    public void consumeAnnotationWithDeprecatedParamsThrowsIllegalArgumentException() {
        // Noop - we expect IllegalArgumentException to be thrown on application startup
    }
```

File: extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedEndpointInjectTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class CamelMainDeprecatedEndpointInjectTest {
+    @RegisterExtension
+    static final QuarkusUnitTest CONFIG = new QuarkusUnitTest().setExpectedException(IllegalArgumentException.class)
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class));
+
+    @EndpointInject(uri = "direct:start")
+    DirectEndpoint start;
+
+    @Test
+    public void camelMainEventBridgeNotConfigured() {
```
Comment (jamesnetherton) [inline extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedEndpointInjectTest.java:37]: See my previous comment about the empty test method.

File: extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedProduceTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+public class CamelMainDeprecatedProduceTest {
+    @RegisterExtension
+    static final QuarkusUnitTest CONFIG = new QuarkusUnitTest().setExpectedException(IllegalArgumentException.class)
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class));
+
+    @Produce(uri = "direct:start")
+    FluentProducerTemplate produceProducerFluent;
+
+    @Test
+    public void camelMainEventBridgeNotConfigured() {
```
Comment (jamesnetherton) [inline extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedProduceTest.java:37]: See my previous comment about the empty test method.

File: extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedConsumeTest.java (hunk 1 above)
Comment (JiriOndrusek) [reply extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedConsumeTest.java]: fixed.

File: extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedEndpointInjectTest.java (hunk 2 above)
Comment (JiriOndrusek) [reply extensions-core/core/deployment/src/test/java/org/apache/camel/quarkus/core/deployment/main/CamelMainDeprecatedEndpointInjectTest.java]: fixed

**Comment by JiriOndrusek** [comment]
@jamesnetherton Would it be possible to merge this fix also to `2.13.x`. To make sure it makes into any patch which may be released?

**Review by jamesnetherton — APPROVED** [review]

**Pull request merged**